tests/integration/test_charm.py

```diff
@@ -35,6 +35,34 @@
 DEFAULT_NUM_UNITS = 2
 
 
+@pytest.mark.group("test-single-node-removal")
```

Comment:
Let's please not create a new group for cases like this. We cannot spin this many workers for each PR run. 
If really needed and we have more than 1 case to test for app removal, please create a new test module: `test_application_removal.py` and move all application removal tests on it (by deploying multiple apps with the constraints you're interested in)

Reply:
okay, it will create and destroy that single-node setup  on the same run. Then the test can proceed normally